lms/djangoapps/shoppingcart/tests/test_views.py

```diff
@@ -531,9 +531,9 @@ def test_soft_delete_coupon(self):
         get_coupon = Coupon.objects.get(id=1)
         request = HttpRequest()
         request.user = admin
-        setattr(request, 'session', 'session')
+        request.session = 'session'
         messages = FallbackStorage(request)
-        setattr(request, '_messages', messages)
+        request._messages = messages        # pylint: disable=protected-access
```

Comment:
Nit: I would generally favor allowing protected access module-wide in test code (and only in test code).